src/pyclass/create_type_object.rs

```diff
@@ -32,22 +32,37 @@ where
     T: PyClass,
 {
     unsafe {
-        PyTypeBuilder::default()
-            .type_doc(T::doc(py)?)
-            .offsets(T::dict_offset(), T::weaklist_offset())
-            .slot(ffi::Py_tp_base, T::BaseType::type_object_raw(py))
-            .slot(ffi::Py_tp_dealloc, tp_dealloc::<T> as *mut c_void)
-            .set_is_basetype(T::IS_BASETYPE)
-            .set_is_mapping(T::IS_MAPPING)
-            .set_is_sequence(T::IS_SEQUENCE)
-            .class_items(T::items_iter())
-            .build(py, T::NAME, T::MODULE, std::mem::size_of::<T::Layout>())
+        PyTypeBuilder {
+            slots: Vec::new(),
+            method_defs: Vec::new(),
+            getset_builders: HashMap::new(),
+            cleanup: Vec::new(),
+            tp_base: T::BaseType::type_object_raw(py),
+            tp_dealloc: tp_dealloc::<T>,
+            tp_dealloc_with_gc: tp_dealloc_with_gc::<T>,
+            is_mapping: T::IS_MAPPING,
+            is_sequence: T::IS_SEQUENCE,
+            has_new: false,
+            has_dealloc: false,
+            has_getitem: false,
+            has_setitem: false,
+            has_traverse: false,
```

Comment:
Maybe we could simplify this using splat syntax `..Default::default()` or is spelling this out intentional?

Reply:
(Ah sorry, you remove the `Default` impl...)